### DISCUSSION TITLE
Reduce the scope of our releases and release the first version

### DISCUSSION THREAD

**Discussion by sleipnir** [eigr/massa#105] [Releases] [answered]
Hello guys I suggest we prepare our first release presenting to the public initially in this first version only the Stateless support with our Action Protocol.
I believe that with this we can attract more people to our project as well as create the habit of releasing incremental releases.
For this to happen we need to focus on:

1. Mature our Operator

2. Define a release method

3. Test the Action Protocol (Forwards and Side-Effects would be left out at this point).

4. Improve our documentation (@marcellanz  has already started an excellent work)

5. Try to run this first release in weeks rather than months

WDYT?

**Comment by wesleimp** [comment]
I think that makes sense. It would be nice to have a release and collect some feedback from users.

> 2. Define a release method

@sleipnir my only question is. What do you mean by "Define a release method"?

**Reply by sleipnir** [reply]
@wesleimp thanks for the answer. In fact I was very vague there, and really in my mind it means a lot of things. I will try to summarize. 
First we need a pipeline that generates the docker images in some registry container, I think @marcellanz created a registry but we never got to integrate in the pipeline. 
The second issue is that we need a way to easily distribute the user to install eigr functions. Currently the manifests are generated in the permastate-operator repo, we need to describe to the user how he should install the manifests. As well, we should check if we have a pipeline generating these manifests so that we can distribute them. 
And finally we must try to understand, otherwise it would be easier to create a CLI that would automate for the user many of the operator installation tasks and also assist in the creation of entity function CRDs

**Reply by wesleimp** [reply]
I gotcha. I've been working on a tool for automated releases, which is on an experimental stage yet, but I think we could do some changes to support docker builds and publishing. This tooling includes changelog generation, semantic versioning based on git tags, GitHub releases, and more. https://github.com/release-cli/release

**Reply by sleipnir** [reply]
@wesleimp  I'm following this project of yours and I found it very interesting, do you think we can integrate it with github actions?

**Reply by wesleimp** [reply]
Yes, I think it's possible. I'm planning to create a specific action to make it easy. Maybe I could finish `v1` in the next weeks.

**Reply by sleipnir** [reply]
Cool ! Let me know about your progress

**Comment by marcellanz** [comment, marked as answer]
@sleipnir makes a lot of sense, thanks! A reduced scope for v0.1.0 enables us to focus and touch all edges of a release too. Regarding docker registry, I created on on gcr.io/eigr-io/ that I support (pay) but if github itself has a registry we can use, we could use it too. I'm not sure if its free or limited in any kind? @wesleimp @sleipnir do you know more?

For the milestone we can change the existing one: https://github.com/eigr/massa/milestone/1 probably adapt to a decision lead by this discussion.

For tasks I did some small work over the new-year, mainly cosmetic fixes and test with the action container we created here `gcr.io/eigr-io/eigr-go-tck-action ` (same as mentioned on the current docs). I'd volunteer for more docs, and I'd create some fill-in-pages for stuff we'll release with v0.1.0. With the Action Protocol; I think in the same direction of reducing the scope, the TCK can be omitted for now. `grpcurl` based examples might be enough for a first release. 

Still, I'm interested in the TCK and would go forward with my work there; one issue I found was, that the proxy was not fast enough to `discover` a running TCK before it gave up with a timeout.

I'm looking forward to collaborate more on eigr.io this year 🚀… let's rock.

**Comment by marcellanz** [comment]
@ralphlaude I assume you can agree, please let us know what you think otherwise :) After that I'd document the tasks in a milestone.

**Reply by ralphlaude** [reply]
I agree with all the propositions. @marcellanz :) you can move forward with the milestone

**Reply by marcellanz** [reply]
thanks @ralphlaude

**Comment by marcellanz** [comment]
@sleipnir @ralphlaude @wesleimp I'll close this discussion. I'll update the milestone.

**Reply by sleipnir** [reply]
Ok @marcellanz
